consensus/consensusfsm/fsm.go

```diff
@@ -321,6 +321,16 @@ func (m *ConsensusFSM) prepare(_ fsm.Event) (fsm.State, error) {
 		return sPrepare, err
 	}
 	m.ctx.Logger().Info("Start a new round", zap.Duration("delay", delay))
+	isProposer := m.ctx.IsProposer()
+	var blk Endorsement
+	if isProposer {
+		m.ctx.Logger().Info("current node is the proposer")
+		if blk, err = m.ctx.MintBlock(); err != nil || blk == nil {
+			// TODO: review the return state
+			m.ProducePrepareEvent(0)
+			return sPrepare, errors.Wrap(err, "error when minting a block")
+		}
+	}
 	if delay > 0 {
```

Comment:
Delete delay?

Reply:
it is still needed to make sure everyone start the consensus at the same time, except the proposer could start mint before that.